kani-driver/src/concrete_playback.rs

```diff
@@ -146,11 +149,13 @@ impl KaniSession {
         drop(temp_writer);
 
         // Renames are usually automic, so we won't corrupt the user's source file during a crash.
-        fs::rename(temp_file.path(), source_path).with_context(|| {
+        fs::rename(&temp_path, source_path).with_context(|| {
             format!("Couldn't rename tmp source file to actual src file `{source_path}`.")
         })?;
 
-        // temp file gets deleted automatically by the NamedTempFile handler
+        drop(temp_file);
+        remove_file(temp_path)?;
```

Comment:
This will only remove the file if the function succeeds though.

Reply:
Yes, thanks for catching that. Is the use of drop here valid? I don't know if explicitly calling drop was good practice or not but I thought it's better to be explicit than assume that it would be dropped when going out of scope.

Reply:
Usually you should only invoke drop if you need it to happen at a certain time or order. I guess you could also manually invoke drop if something is taking a lot of memory. Otherwise, just leave it to the compiler. I think in this case you do want to drop `temp_file` before removing the file. You could wrap these two things into one object and drop everything in the correct order as part of its drop logic.